mautic/mautic
# Ip service lookup changes

**Description**

This PR addresses and fixes #1231.  The default IP lookup service, Telize, shut down. This PR does several things:

1. It now defaults to a newly added MaxMind GeoIp2 City Download service.  This will download Maxmind's free city IP lookup database which will be used to glean general info on an IP.
2. To support such an IP lookup with accompanying form UI elements, the IpService classes have been re-organized while keeping BC support for old AbstractIpLookup use in case any 3rd party IP lookup services exist out in the wild
3. `Mautic\CoreBundle\IpLookup\AbstractLookup` replaces the deprecated `AbstractIpLookup`
3. `Mautic\CoreBundle\IpLookup\AbstractLocalDataLookup` has been added with accompanying methods for data stores that are stored locally (such as above)
4. `Mautic\CoreBundle\IpLookup\AbstractRemoteDataLookup` has been added with accompanying methods for services that use API calls to fetch IP details
5. `Mautic\CoreBundle\IpLookup\IpLookupFormInterface` has been added for services that support injecting custom config options into Mautic's configuration
6. `Mautic\CoreBundle\Factory\IpLookupFactory` has been added to create the different lookup services when necessary
7. `mautic:iplookup:download` command has been added to allow use of cron to keep local datastores up to date
8. PhpUnit was added as a dev dependency and config_test.php was slightly changed to support the basic tests included in the PR
9. Telize was converted to use it's now for pay API

**Testing**

1. Apply the PR
2. Run `composer install`
3. Run the tests if you have PhpUnit configured on your system
4. Or manually test by selecting each IP service and test by going to a public landing page on a remote Mautic system (can't be 127.0.0.1 or ::1), get details, delete details in the ip_addresses table, then try the next
5. Choose the MaxMind GeoIp2 City Download service a button should appear to download the file.  It'll be btn-danger if the file hasn't been downloaded with "Fetch..." or btn-success if it has been with "Update..." and a tooltip of when the file was last updated
6. Click the button and should get a success response after the data store has been downloaded
7. Ensure the config saves
